signer/src/storage/model.rs

```diff
@@ -84,6 +91,17 @@ impl From<&crate::message::SweepTransactionInfo> for SweepTransaction {
     }
 }
 
+impl utxo::GetFees for Vec<SweepTransaction> {
+    /// Return the total fee of all the transactions in the vector.
+    fn get_fees(&self) -> Fees {
+        let total = self.iter().map(|tx| tx.fee).sum();
+        let total_size: u32 = self.iter().map(|tx| tx.vsize).sum();
+        let rate: f64 = total as f64 / total_size as f64;
```

Comment:
should we add a `total_size == 0` check here? I guess it shouldn't happen, merely as a safety guard

Reply:
Right, I don't think it's even possible to create a zero-sized `bitcoin::Transaction` -- but I'll toss that in while I'm doing the above refactor just so it's 100% sound.

Reply:
That too, so good catch

Reply:
Added in [8018721](https://github.com/stacks-network/sbtc/pull/810/commits/8018721e7104fdbe5b368af70835ffbc23f437c5)